### PR TITLE
[xaudio2redist] port updated to version 1.2.9

### PR DESCRIPTION
**Pull request by walbourn**
Update port for latest release of ``Microsoft.XAudio2.Redist`` on NuGet:

* When Virtual Surround Sound (VSS) is enabled on an audio endpoint, the Mastering Voice opens the audio endpoint in 7.1 channel mode.

### CODE REVIEW

**Comment by walbourn** [comment]
Validated supported triplets:

```
vcpkg install xaudio2redist:x64-windows
vcpkg install xaudio2redist:x86-windows
vcpkg install xaudio2redist:x64-windows-static-md
vcpkg install xaudio2redist:x86-windows-static-md

vcpkg install directxtk[xaudio2redist]:x64-windows --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows --recurse
vcpkg install directxtk[xaudio2redist]:x64-windows-static-md --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows-static-md --recurse
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@walbourn, 
````
vcpkg_extract_source_archive was passed extra arguments: PACKAGE_PATH
````
Please restore the use of vcpkg_extract_source_archive_ex, we don't want to simply update to introduce too many other problems

File: ports/xaudio2redist/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://www.nuget.org/api/v2/package/Microsoft.XAudio2.Redist/1.2.8"
-    FILENAME "xaudio2redist.1.2.8.zip"
-    SHA512 509b2783457b86ed1878fd4e14a01fa7288591925a2bb3cad4d68afd597fbff1f1349b619dad628b5d685169825a775120e1611559e9097837cff0fb6d39acf0
+    URLS "https://www.nuget.org/api/v2/package/Microsoft.XAudio2.Redist/${XAUDIO2REDIST_VERSION}"
+    FILENAME "xaudio2redist.${XAUDIO2REDIST_VERSION}.zip"
+    SHA512 c3b37640fb871523a63cd227653d8d972dd95d6e12ccf2f28c434f51bb77011c821a0cd5ae2a9fa311f005a0083798a3218a98c0a9db5db094a5ef54bb960675
 )
 
 vcpkg_extract_source_archive_ex(
```
[the remark this answers is not in the document]
Comment (JonLiu1993) [reply ports/xaudio2redist/portfile.cmake:11]: @walbourn ,I restore the use of vcpkg_extract_source_archive_ex, I don't want to simply update to introduce too many other problems

**Review by JonLiu1993 — APPROVED** [review]

File: ports/xaudio2redist/portfile.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (walbourn) [reply ports/xaudio2redist/portfile.cmake:11]: Sorry, I should have tried it locally...  The change to make it work is:

1.`vcpkg_extract_source_archive_ex` to `vcpkg_extract_source_archive`.
2. Remove `OUT_SOURCE_PATH`

[the remark this answers is not in the document]
Comment (walbourn) [reply ports/xaudio2redist/portfile.cmake]: Fixed and verified.

**Comment by walbourn** [comment]
Revalidated supported triplets after code review:

```
vcpkg install xaudio2redist:x64-windows
vcpkg install xaudio2redist:x86-windows
vcpkg install xaudio2redist:x64-windows-static-md
vcpkg install xaudio2redist:x86-windows-static-md

vcpkg install directxtk[xaudio2redist]:x64-windows --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows --recurse
vcpkg install directxtk[xaudio2redist]:x64-windows-static-md --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows-static-md --recurse
```

**Review by ras0219-msft — APPROVED** [review]
Thanks for the update!

**Pull request merged**